public/app/plugins/datasource/loki/dataProviders/logsVolumeProvider.ts

```diff
@@ -0,0 +1,127 @@
+import {
+  DataFrame,
+  DataQueryRequest,
+  DataQueryResponse,
+  FieldCache,
+  FieldColorModeId,
+  FieldType,
+  LogLevel,
+  LogsVolume,
+  QueryRelatedDataProvider,
+  toDataFrame,
+} from '@grafana/data';
+import { LokiQuery } from '../types';
+import { Observable, SubscriptionLike } from 'rxjs';
+import { cloneDeep } from 'lodash';
+import LokiDatasource, { isMetricsQuery } from '../datasource';
+import { aggregateFields, getLogLevelFromLabels } from '../../../../features/explore/state/utils';
+import { LogLevelColor } from '../../../../core/logs_model';
+import { BarAlignment, GraphDrawStyle, StackingMode } from '@grafana/schema';
+
+export class LokiLogsVolumeProvider implements QueryRelatedDataProvider<LogsVolume> {
+  private readonly datasource: LokiDatasource;
+  private dataQueryRequest: DataQueryRequest<LokiQuery>;
+  private rawLogsVolume: DataFrame[] = [];
+  private currentSubscription?: SubscriptionLike;
+
+  constructor(datasource: LokiDatasource) {
+    this.datasource = datasource;
+  }
+
+  setRequest(dataQueryRequest: DataQueryRequest<LokiQuery>): void {
+    this.dataQueryRequest = dataQueryRequest;
+  }
+
+  getData(): Observable<LogsVolume> {
+    if (this.currentSubscription) {
+      this.currentSubscription.unsubscribe();
+      this.currentSubscription = undefined;
+    }
+
+    const histogramRequest = cloneDeep(this.dataQueryRequest);
+    histogramRequest.targets = histogramRequest.targets
+      .filter((target) => !isMetricsQuery(target.expr))
+      .map((target) => {
+        target.expr = `count_over_time(${target.expr}[$__interval])`;
```

Comment:
Can we already do `by level` or something here so we don't have to manually aggregate later on?

Reply:
When count_over_time is created we don't know what label is used for level but I think we could add this to data source configuration.

Reply:
Ah I see now that we get all the labels and then just check different ones if they exist. Yeah we could add a config maybe that would be optional, ie if it is set use that directly in the query other wise fallback to what we have. But maybe we should check if this is actually any issue performance wise or not first.